File: client/bcephal-client-model/Forms/FormData.cs
using Bcephal.Models.Base;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bcephal.Models.Forms
{
    public class FormData : Persistent
    {

        public long? ModelOid { set; get; }

        public long? FormModelId { set; get; }

        public long? GridId { set; get; }

        public int Position { set; get; }

        public bool Validated { set; get; }

        public Dictionary<long?, FormDataValue> Datas { set; get; }

        public Dictionary<long?, ListChangeHandler<FormData>> SubGridDatas { set; get; }

        public FormData()
        {
            this.Datas = new Dictionary<long?, FormDataValue>(0);
            this.SubGridDatas = new Dictionary<long?, ListChangeHandler<FormData>>(0);
            this.Validated = false;
        }

        public override int CompareTo(object obj)
        {
            if (obj == null || !(obj is FormData)) return 1;
            if (this == obj) return 0;
            return this.Position.CompareTo(((FormData)obj).Position);
        }


    }
}

File: server/bcephal-grid/src/main/java/com/moriset/bcephal/grid/domain/form/FormData.java
/**
 * 
 */
package com.moriset.bcephal.grid.domain.form;

import java.util.HashMap;
import java.util.Map;

import com.moriset.bcephal.domain.ListChangeHandler;
import com.moriset.bcephal.domain.MainObject;
import com.moriset.bcephal.domain.Persistent;

import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * @author Joseph Wambo
 *
 */
@Data
@EqualsAndHashCode(callSuper = false)
public class FormData extends MainObject {

	private static final long serialVersionUID = -322354323554458862L;

	private Long id;
	private Long modelOid;
	private Long formModelId;
	private Long gridId;
	private int position;
	private boolean validated;

	private Map<Long, FormDataValue> datas;

	private Map<Long, ListChangeHandler<FormData>> subGridDatas;

	public FormData() {
		datas = new HashMap<>();
		subGridDatas = new HashMap<>();
	}

	@Override
	public Persistent copy() {
		return null;
	}

}
